aggregator/src/events/events_store.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{
+    common::{
+        constants::{
+            EVENT_STATS_DATA, EVENT_STORE, EVENT_STORE_LABLE_KEY, PATCH_PARAM_FILED_MANAGER,
+        },
+        error,
+        utils::get_release_name,
+    },
+    Cache, EventSet,
+};
+use k8s_openapi::api::core::v1::ConfigMap;
+use kube::{
+    api::{Api, Patch, PatchParams, PostParams},
+    core::ObjectMeta,
+    Client,
+};
+use snafu::ResultExt;
+use std::{collections::BTreeMap, ops::DerefMut, time::Duration};
+use tracing::info;
+
+/// Initialize a config map for storing events.
+pub async fn initialize(namespace: &str) -> error::Result<ConfigMap> {
+    let client = Client::try_default().await.context(error::K8sClient)?;
+    let release_name = get_release_name(namespace, client.clone()).await?;
+    let api: Api<ConfigMap> = Api::namespaced(client.clone(), namespace);
+    let config_map_name = format!("{release_name}-{EVENT_STORE}");
+
+    if let Some(cm) =
+        api.get_opt(config_map_name.as_str())
+            .await
+            .context(error::GetEventStoreConfigMap {
+                name: config_map_name.clone(),
+            })?
+    {
+        info!(
+            "Config map {} for events store already exists.",
+            config_map_name.clone()
+        );
+        return Ok(cm);
+    }
+
+    info!(
+        "Creating Config map {} for events store!!!",
+        config_map_name.clone()
+    );
+    let cm = create_configmap(namespace, client.clone(), config_map_name.as_str()).await?;
```

Comment:
IMO this codebase shouldn't create the configmap. If this component creates the configmap, then this module will have to handle upgrades for it, kind of like how the diskpool operator handles CRD upgrades today. It might be a good idea to defer this responsibility to the upgrade components. Maybe the helm chart should create it. The helm chart will also allow for cleanup on uninstall... wdyt @sinhaashish @datacore-vvarakantham @tiagolobocastro @avishnu @chriswldenyer?

Reply:
Yep it makes sense I think.

Reply:
done. PR https://github.com/openebs/mayastor-extensions/pull/255